MSAL/src/native_auth/controllers/credentials/MSALNativeAuthCredentialsController.swift

```diff
@@ -171,7 +171,7 @@ final class MSALNativeAuthCredentialsController: MSALNativeAuthTokenController,
         telemetryEvent: MSIDTelemetryAPIEvent?,
         context: MSALNativeAuthRequestContext,
         config: MSIDConfiguration
-    ) -> Result<String, RetrieveAccessTokenError> {
+    ) -> RefreshTokenCredentialControllerResponse {
         do {
             let tokenResult = try cacheTokenResponse(tokenResponse, context: context, msidConfiguration: config)
             telemetryEvent?.setUserInformation(tokenResult.account)
```

Comment:
From my understanding, the telemetryEvent should no longer be stopped on the following line because the operation might yet be unsuccessful, and we are already calling the new `updateTelemetryEvent` later in the flow (line 184). This is the same in a few places in the new code.

Reply:
That's a good catch, thank you. I've reviewed every controller, and found that it was happening in other places as well.